packages/app/src/cli/models/extensions/schemas.ts

```diff
@@ -32,17 +37,54 @@ export const ApiVersionSchema = zod.string()
 
 export type ApiVersionSchemaType = zod.infer<typeof ApiVersionSchema>
 
+export const SettingsSchema = zod.object({
+  fields: zod
+    .array(
+      zod.object({
+        key: zod.string().optional(),
+        name: zod.string().optional(),
+        description: zod.string().optional(),
+        required: zod.boolean().optional(),
+        type: zod.string(),
+      }),
+    )
+    .optional(),
+})
+
+export const HandleSchema = zod
+  .string()
+  .trim()
+  .min(3, 'Handle needs to have at least 3 characters')
+  .max(30, "Handle can't exceed 30 characters")
+  .regex(/^[a-zA-Z0-9.\-_]*$/, 'Handle can only contain alphanumeric characters, periods, hyphens, and underscores')
```

Comment:
Updated guidance for handle validation (only alphanumeric and hyphens): https://github.com/Shopify/shopify/pull/435644#issuecomment-1629915032

Reply:
Makes sense to me 🤷